tests/v2/validation/extensions/provisioning/permutations/permutations.go

```diff
@@ -0,0 +1,224 @@
+package permutations
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/rancher/rancher/tests/framework/clients/corral"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/machinepools"
+	"github.com/rancher/rancher/tests/framework/extensions/provisioning"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/v2/validation/extensions/provisioning/crud"
+	nodeProviders "github.com/rancher/rancher/tests/v2/validation/extensions/provisioning/providers"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	RKE2CustomCluster    = "rke2Custom"
+	RKE2ProvisionCluster = "rke2"
+	RKE2AirgapCluster    = "rke2Airgap"
+	K3SCustomCluster     = "k3sCustom"
+	K3SProvisionCluster  = "k3s"
+	K3SAirgapCluster     = "k3sAirgap"
+	RKE1CustomCluster    = "rke1Custom"
+	RKE1ProvisionCluster = "rke1"
+	RKE1AirgapCluster    = "rke1Airgap"
+)
+
+// RunTestPermutations runs through all relevant perumutations in a given config file, including node providers, k8s versions, and CNIs
+func RunTestPermutations(s *suite.Suite, testNamePrefix string, client *rancher.Client, clustersConfig *provisioning.Config, clusterType string, hostnameTruncation *machinepools.HostnameTruncation, corralPackages *corral.CorralPackages) []string {
+	var name string
+	var clusterNames []string
+	// not currently returned, but could use in the future to optimize tests via saving audits for after cluster creation
```

Comment:
We could probably omit this comment. There is a chance that we forget to remove this comment in the future when it's no longer relevant, and it may throw some confusion into the pot.